### PR TITLE
ci: use aws codebuild

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

Following:
- https://docs.aws.amazon.com/codebuild/latest/userguide/action-runner.html

[Relevant machines:](https://aws.amazon.com/codebuild/pricing/)
| Instance Type    | Memory | vCPUs | Price per Minute |
|------------------|--------|-------|----------------|
| arm1.large       | 16 GiB | 8     | $0.015         |
| general1.medium  | 7 GB   | 4     | $0.01          |
| general1.large   | 15 GB  | 8     | $0.02          |
| general1.xlarge  | 72 GiB | 36    | $0.0798        |

- all tests will use `total cpu count - 1` cores to build, to allow runner to function.
   - right now the taskset cpu count is provided manually, can be made into a simple workflow step in the future
   - otherwise machines lose connection
   - we make use of `taskset` from GNU coreutils for this job
- ~~**build-and-test** will use `general1.medium` since it is not time critical and it's ok for it to be slow~~
   - It exceeded 2 hours with just 3 cores so switching it to large now.
- **build-and-test-daily-arm64** will use `arm1.large`
   - it runs once per day so it's alright to have large machine
- **build-and-test-differential-arm64** will use `arm1.large`
   - this workflow requires arm tag and is not required for merging, will rarely be used
- **build-and-test-differential** will use `general1.large`
   - in the future we can optimize it so that:
      - building is done on `general1.xlarge`
         - building is parallel-friendly and we are charged per cpu/minutes so this would make it finish much faster with similar price
      - test and codecov upload is done on a `general1.medium`
         - tests are running sequentially, more than 4 cores is wasted with these tests
         - codecov could even run on github hosted runners

## Related links

- https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-compute-types.html
- https://aws.amazon.com/codebuild/pricing/

**Private Links:**

- https://us-east-2.console.aws.amazon.com/codesuite/codebuild/305110990523/projects/autoware-codebuild/history?region=us-east-2&builds-meta=eyJmIjp7InRleHQiOiIiLCJzdGF0dXMiOiIifSwicyI6e30sIm4iOjIwLCJpIjowfQ
- https://github.com/autowarefoundation/autoware.universe/settings/actions/runners
- https://us-east-1.console.aws.amazon.com/servicequotas/home/services/codebuild/quotas

## Tests

- **build-and-test**: https://github.com/autowarefoundation/autoware.universe/actions/runs/10469798912 ❌
   - exceeded 2 hours with just 3 cores.
   - Will switch it to large machine now.
   - https://github.com/autowarefoundation/autoware.universe/actions/runs/10471948833/job/29000306215
- **build-and-test-daily-arm64**: https://github.com/autowarefoundation/autoware.universe/actions/runs/10469818933
   - build succeeded ✅
   - test fails but it is not related to this pr, arm tests have been failing for a while.
- **build-and-test-differential**: https://github.com/autowarefoundation/autoware.universe/actions/runs/10469689956 ✅
- **build-and-test-differential-arm64**: https://github.com/autowarefoundation/autoware.universe/actions/runs/10469689962 ✅

### Old tests

https://docs.google.com/spreadsheets/d/1hO7rxgq_cYAoAEc7EWOFs-e_fwxeTwTJOBznAtBkPac/edit?usp=sharing

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/actions/runs/10417445920/job/28851709384?pr=8498

concurrency fixed the "old runners keep running for the older commits" problem.

- https://www.meziantou.net/how-to-cancel-github-workflows-when-pushing-new-commits-on-a-branch.htm
- https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-using-concurrency-and-the-default-behavior

File: sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp [hunk 1]
```diff
@@ -85,6 +85,8 @@
 #include <autoware/universe_utils/ros/static_transform_buffer.hpp>
 #include <autoware/universe_utils/system/stop_watch.hpp>
 
+// TODO(xmfcx): remove this line before merging...
+
```
Comment (xmfcx) [inline sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp:89]:
```suggestion
```

To be removed before merging.

**Review by mitsudome-r — APPROVED** [review]

**Comment by xmfcx** [comment]
@mitsudome-r I've merged it. If something goes wrong, feel free to revert this change.

But before reverting, please link the failing CI reports under here and try to rerun them at least once.

**Comment by isamu-takagi** [comment]
@xmfcx @mitsudome-r 

It seems that the README added in this PR overrides the README in the repository root.
https://github.com/autowarefoundation/autoware.universe?tab=readme-ov-file

>If a repository contains more than one README file, then the file shown is chosen from locations in the following order: the .github directory, then the repository's root directory, and finally the docs directory.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes

**Pull request merged**